### ISSUE TITLE
v1.66 and earlier, after  Kodi v19.0 Version Upgrade - Advanced Editor - Add-on Error 

### ISSUE THREAD

**Issue by sammydc77** [robweber/xbmcbackup#187]
Before working on a new issue I'm probably going to ask you all this stuff anyway, probably just easier to provide it now.....

Hello:
This is the related information.  Let me know if you need additional.

Regards,

Joseph
______________________________________

**Describe the problem**

Every time I try [Advanced Editor], Kodi responds with - [Add-on error.  Check the log for more information], either from Configuration or running the Plugin.

**Platform and Kodi version**
What OS and Kodi version are you using? (Rpi, Android, etc)

Google Pixelbook i5 8GB / 256GB, with External Silicon Power B75 Pro 1TB SSD;  Chrome OS Current Stable State [Version 88.0.4324.208 (Official Build) (64-bit)].

2021-03-20 11:04:48.394 T:13364    INFO <general>: Starting Kodi (19.0 (19.0.0) Git:20210218-f44fdfbf67). Platform: Android ARM 64-bit
2021-03-20 11:04:48.394 T:13364    INFO <general>: Using Release Kodi x64
2021-03-20 11:04:48.395 T:13364    INFO <general>: Kodi compiled 2021-02-18 by Clang 8.0.7 (https://android.googlesource.com/toolchain/clang b55f2d4ebfd35bf643d27dbca1bb228957008617) (https://android.googlesource.com/toolchain/llvm 3c393fe7a7e13b0fba4ac75a01aa683d7a5b11cd) for Android ARM 64-bit API level 21 (API level 21)

**Link to Debug Log**
Don't just post the whole thing here,  use [https://paste.kodi.tv](https://paste.kodi.tv)

https://paste.kodi.tv/mivoboxazo.kodi

**Comment by robweber** [comment]
My best guess about what is happening here is that the log states it is adding all addons to the Python path since an addon id was not specified as part of the RunScript() command. There is more than likely an overlap in the ```resources.lib``` module path causing an error in finding the required libraries. 

I have an idea for a workaround but it's going to require some refactoring. I can't reproduce it so it's an edge case where a user would have to have an addon with a conflicting path and also be using the advanced editor. Once I have something for you to test I'll let you know.

**Comment by sammydc77** [comment]
Rob:

I understand the issues involved, it is rarely clear-cut, pointing at a
cause.  Please take your time, and most of all enjoy the challenge!

Joseph

**Comment by robweber** [comment]
If you're able can you try the fixes from this version? https://github.com/robweber/xbmcbackup/tree/launcher_refactor 

It's not 100% but I think the tweaks so far will modify the RunScript() method so it doesn't load all your addon libs in the Python PATH. Hopefully that should fix the issue.

**Comment by sammydc77** [comment]
Hello Rob:

I appreciate this and will give it a try.  I will let you know the results.

Joseph

**Comment by robweber** [comment]
Just a note some of the outcome of this is listed in #188

**Comment by sammydc77** [comment]
Rob:

I understand, and expected as much.  By the way, do you have any thought of
a simple experiment I can try to get Kodi v19.0  to see External SSD
Drives, as v18.9 used to do?  Apparently the Kodi Team does not want to
touch this issue.  I will not ask anything further of you.  Surprisingly, I
just did a search on Google and two Sites have my Kodi Bug Report, it seems
this is a real issue.

https://www.google.com/search?q=how+to+make+kodi+19+see+external+ssd&rlz=1CAOTWH_enUS849&oq=how+to+make+kodi+19+see+external+ssd&aqs=chrome..69i57.23433j0j7&sourceid=chrome&ie=UTF-8


Joseph

**Comment by robweber** [comment]
Depending on the device it could be a permissions thing. Perhaps the user account Kodi runs under can't see the mounted device? I've seen this happen on linux and android based systems before. 

I'm going to make the necessary adjustments to make this fix permanent and push with the next release.

**Comment by sammydc77** [comment]
Rob:

I look forward to your next release.  I appreciate you taking the time to
consider the external drive issue.  I had absolutely no problem with Kodi
v18.9, and all prior versions [SPMC, Kodi v17.6...], only after v19.0 was
installed, the remaining plugins [including your Backup], could still see
the external drive, but with UUID/GUID only, no Label.  If I uninstalled
and reinstalled a plugin, it no longer saw the external drive...  Thank
You, for the update and I wish you well.  By the way, the first thing I
check was whether Kodi v19.0 had Storage Rights and the External Drive also
had the  Storage Access enabled, so all installed Android Apps could see
and have full access to it.  I am getting a new External SSD Drive, n a few
days, and have enabled the Linux [Crostini], so I can experiment with the
Linux version of Kodi v19.0, if nothing else it should be fun.

Regards,

Joseph

**Comment by sammydc77** [comment]
Thanks, I'll check it out.

Joseph
